Factor out common implementation of help(obj) into library function

infinnovation-dev opened the pull request:
Several ports use identical code for the 1-argument form of the builtin help function.  Move this code to a library function and update existing help implementations to use it.  This should aid adding help implementations for other ports.

Commit:
Extract implementation of help(obj) to a library function.

Several ports use identical code for the 1-argument form of the builtin
help function.  Move this code to a library function to allow easier
re-use by ports.

Commit:
cc3200, stmhal, teensy: Use pyhelp_print_obj function.

Update the help() implementations in the cc3200, stmhal and teensy
ports to use the pyhelp_print_obj function.

pfalcon commented on lib/utils/pyhelp.h:
```diff
@@ -0,0 +1,33 @@
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016 Colin Hogben
```
What exactly being copyrighted here?

infinnovation replied:
It's a line of code defining an API.  Agreed, it's hardly the most ground-breaking piece of code ever written, but it didn't exist before, and I wrote it, so (subject to jurisdiction, IANAL) I de facto own the copyright.  I didn't see anything about copyright in the contributor guidelines, but I'm happy to assign copyright to Damien or yourself or to the project as a whole if that is preferred.

pfalcon replied:
If you just move code around, please keep the original copyright. If you add some noticeable code and functionality, add your copyright (we need to remind that to people regularly too).

infinnovation replied:
That's why I kept Damien's copyright in pyhelp.c since that was just a minor reshuffle of the original help.c implementations.  What is your preference for new but trivial code such as pyhelp.h?

dpgeorge replied:
Usually I don't put any copyright header in such cases.

pfalcon replied:
> Usually I don't put any copyright header in such cases.

I believe all files in the repo have the same header template, which has copyright statement, including .h files.

> What is your preference for new but trivial code such as pyhelp.h?

It should be the same as original file from where it was split, or just Damien (majority of code in the repo authored by him).

dpgeorge replied:
> Usually I don't put any copyright header in such cases.
> 
> I believe all files in the repo have the same header template, which has copyright statement, including .h files.

There are some .h files that have no header, eg esp8266/modpyb.h, esp8266/espneopixel.h, esp8266/esppwm.h, stmhal/mphalport.h, extmod/modwebsocket.h.

Commit:
pic16bit: Add minimal help() implementation.

The REPL suggests 'Type "help()" for more information.'
Providing at least some clue is better than a NameError.

Commit:
lib/utils: Remove copyright/licence notice

Remove the copyright header from this trivial file as per discussion of PR #2002

pfalcon commented:
Thanks! I've cherrypicked couple of commits and cleaned them up (commit message format/consistency of copyright notices). Feel free to rebase and submit other patches if there's argumentation why they should be included.
